lib/outputs/outputs_test.py

```diff
@@ -254,10 +257,68 @@ def test_fill_json_asff(self):
 
         expected.Compliance = Compliance(
             Status="PASS" + "ED",
-            RelatedRequirements=[finding.check_metadata.CheckType],
+            RelatedRequirements=finding.check_metadata.CheckType,
         )
         expected.Remediation = {
             "Recommendation": finding.check_metadata.Remediation.Recommendation
         }
 
         assert fill_json_asff(input, input_audit_info, finding) == expected
+
+    @mock_s3
+    def test_send_to_s3_bucket(self):
+        # Create mock session
+        session = boto3.session.Session(
+            region_name="us-east-1",
+        )
+        # Create mock audit_info
+        input_audit_info = AWS_Audit_Info(
+            original_session=None,
+            audit_session=session,
+            audited_account="123456789012",
+            audited_identity_arn="test-arn",
+            audited_user_id="test",
+            audited_partition="aws",
+            profile="default",
+            profile_region="eu-west-1",
+            credentials=None,
+            assumed_role_info=None,
+            audited_regions=["eu-west-2", "eu-west-1"],
+            organizations_metadata=None,
+        )
+        # Creat mock bucket
+        bucket_name = "test_bucket"
+        client = boto3.client("s3")
+        client.create_bucket(Bucket=bucket_name)
+        # Create mock csv output file
+        output_directory = "."
+        output_mode = "csv"
+        filename = (
+            f"prowler-output-{input_audit_info.audited_account}-{output_file_timestamp}"
+        )
+        file_descriptor = open_file(
+            f"{output_directory}/{filename}{csv_file_suffix}",
+            "a",
+        )
+        # Send mock csv file to mock S3 Bucket
+        send_to_s3_bucket(
+            filename,
+            output_directory,
+            output_mode,
+            bucket_name,
+            input_audit_info.audit_session,
+        )
+        # Check if the file has been sent by checking its content type
+        assert (
+            client.get_object(
+                Bucket=bucket_name,
+                Key=output_directory
+                + "/"
+                + output_mode
+                + "/"
+                + filename
+                + csv_file_suffix,
+            )["ContentType"]
+            == "binary/octet-stream"
```

Comment:
Why don't we check if the uploaded file match with the generated file?

I prefer to check the file name.